oeps/oep-0014.rst

```diff
@@ -0,0 +1,175 @@
+| Author        | Christina Roberts, christina@edx.org      |
++---------------+-------------------------------------------+
+| Arbiter       | TBD                                       |
++---------------+-------------------------------------------+
+| Status        | Draft                                     |
++---------------+-------------------------------------------+
+| Type          | Process                                   |
++---------------+-------------------------------------------+
+| Created       | 2017-01-09                                |
++---------------+-------------------------------------------+
+| `References`  | `ORA PR Discussion`_,                     |
+|               | `Initial Archiving Discussions`_          |
++---------------+-------------------------------------------+
+
+.. _ORA PR Discussion: https://github.com/edx/edx-ora/pull/187
+.. _Initial Archiving Discussions: https://openedx.atlassian.net/wiki/display/IT/Proposed+Github+Deprecation+Process
+
+Abstract
+========
+
+The `edx organization`_ contains a large number of repositories, most of
+which are active and maintained, but some of which are now obsolete. To
+clarify the status of repositories, and to ensure that we are only monitoring
+active repositories via Gemnasium (which alerts about the usage of third
+party libraries with known security issues), a process for archiving a
+repository is defined below.
+
+.. _edx organization: https://github.com/edx
+
+Motivation
+==========
+
+Recently openedx.yaml files were added to edX repositories per
+`OEP-2 <oep-0002.rst>`_. In the course of deciding owners for those repositories,
+there was a `ORA PR Discussion`_ about how best to handle deprecated or obsolete
+repositories; in particular, do obsolete repositories need owners, and how can
+repositories be clearly marked as present for archive purposes only?
+
+This discussion resurfaced related to edX's usage of Gemnasium to report
+the usage of third-party libraries that have known security issues. All
+repositories under the edX organization were being monitored, but this added
+noise when trying to understand the number of third-party library updates
+required for actively maintained repositories.
+
+Specification
+=============
+
+When a repository under the `edx organization`_ will no longer be maintained
+because it is no longer in use, the following steps should be followed:
+
+First, send an email to edx-code@googlegroups.com announcing that the
+repository will be archived, and inquiring if anyone would like to take
+ownership of the repo. If someone wishes to take ownership, transfer the
+repository to their github organization and hand over rights [Open questions:
+Do we need to outline these steps further? Would there need to be internal
+sign-off before transferring the repo outside of edX? Would we wish to create
+our own fork of the repo?].
+
+If nobody wishes to take ownership of the repository then:
```

Comment:
Ah, I hadn't seen that page. Helpful context. I don't feel very strongly about this, but I think I like Option 1 the best.

The current proposal feels like it optimizes for the less frequent use case (searching and seeing history) than for the most frequent use case (searching and seeing code under active development). I also don't see a significant difference in the amount of maintenance work and overhead.

That is I would rather have to opt-in to seeing history than to opt-out of seeing it. Particularly given the fact that as far as I can tell, when you perform a code search on github (as opposed to a repository search) it doesn't appear to show the description of the repo in the search result, see:

![screenshot from 2017-01-10 07-41-08](https://cloud.githubusercontent.com/assets/627177/21806611/4e52e4bc-d708-11e6-96bf-0643ce19a2fa.png)

This is in the results for the search "ora user:edx". In order to also opt-in to searching the archive, you could simply search for "ora user:edx user:edx-archive". In this example, it isn't abundantly clear what the difference is between "edx-ora" and "edx-ora2".

It also looks like the results are ordered by "best match" not by repo activity, so this could easily return deprecated repos above active ones.